### ISSUE TITLE
How to running

### ISSUE THREAD

**Issue by fajarwahyu99** [riochr17/Analisis-Sentimen-ID#1]
Bagaimana cara run program ini?
Apakah bisa digunakan untuk skripsi?

**Comment by riochr17** [comment]
halo @fajarwahyu99 
untuk menjalankannya pastikan python kamu sudah terpasang library berikut:
- scikit-learn, 
- nltk, 
- numpy, 
- keras (dan backendnya), dan 
- csv

**Comment by fajarwahyu99** [comment]
ada tutorial untuk pasang semua library itu tidak?

**Comment by riochr17** [comment]
untuk menjalankan pengujian kamu bisa lakukan perintah berikut:
https://github.com/riochr17/Analisis-Sentimen-ID/blob/516d11ba66002cf6580ae4598e980ca71501df0a/main.py#L117-L124

untuk melakukan pelatihan data silahkan uncomment baris berikut
https://github.com/riochr17/Analisis-Sentimen-ID/blob/516d11ba66002cf6580ae4598e980ca71501df0a/main.py#L111-L115

**Comment by riochr17** [comment]
bisa pakai pip (package manager) untuk python, googling saja nanti ada tutorialnya

**Comment by faqisyadid** [comment]
Ka mau tanya itu data training nya kita yg beri label sendiri atau otomatis itu ya ka?

**Comment by riochr17** [comment]
data trainingnya ada disini [data/training_all_random.csv](https://github.com/riochr17/Analisis-Sentimen-ID/blob/master/data/training_all_random.csv) sudah diberi label `1` untuk positif, `0` untuk negatif

**Comment by faqisyadid** [comment]
Brrti itu labelin nya manual ya ka, klw untuk model h5 itu gunanya untuk apa ya ka?

**Comment by riochr17** [comment]
h5 itu utk menyimpan model weights dari multilayer perceptron yang sudah ditrain

**Comment by faqisyadid** [comment]
ka mau tanya itu  di preprocess itu gk ada stopwordnya ya ka?

saya juga msh bingung ka dengn kegunaan fungsi kbbi itu mksdnya gmn ? soalnya tadi saya panggil fungsi itu tidak jalan

**Comment by riochr17** [comment]
kbbi itu utk normalisasi kata2 yang non-formal, seperti ini
https://github.com/riochr17/Analisis-Sentimen-ID/blob/516d11ba66002cf6580ae4598e980ca71501df0a/kamus/kbba.txt#L1-L20

preproses stopword blm ada, bisa kamu tambahkan, nnti pull request, saya coba gabung.

**Comment by faqisyadid** [comment]
ka mau tanya lagi, saya blom paham dari kaksud code ini gmn ya ka? di bagian kelas train. indim itu kegunaan apa? dan anga 0.39 itu untuk apa ya ka?

indim = len(X[0])
	indim = 300 if indim > 300 else indim

	model.add(Dense(units=int(0.39 * indim), activation='tanh', input_dim=indim))

**Comment by riochr17** [comment]
@faqisyadid 
indim itu representasi dimensi input, disini saya membatasi maksimal input dimensi sebesar 300, sehingga:
```layer pertama sebesar <indim> node```
```layer kedua sebesar <indim> * 0.39 node```
```layer ketiga sebesar 5 node ```
```layer keempat / ouput sebesar 1 node```

angka 0.39 hanya heuristik untuk eksperimen, bisa diganti dengan angka lain


https://github.com/riochr17/Analisis-Sentimen-ID/blob/516d11ba66002cf6580ae4598e980ca71501df0a/main.py#L28-L33

**Comment by riochr17** [comment]
o iya utk pertanyaan lain, silahkan buka issue baru ya, terima kasih.
